### ISSUE TITLE
Many files get lost, please help. 

### ISSUE THREAD

**Issue by chaos369** [chrislusf/seaweedfs#1916]
**Describe the bug**
The steps of operation:
1 Export the mate data from the filer, because it's stored in local file. 
2 Inport the mate data to redis
3 change master ip, filer ip from public ip address to Lan ip address.
4 added two Volume nodes and one filer node on anther server

Then I found many files get lost. It printed warning log :
E0317 16:08:05 37939 stream.go:42] read chunk: fetch http://[2a01:4f8:236:60d3::9]:9090/6,01180d63797c8e?readDeleted=true: http://[2a01:4f8:236:60d3::9]:9090/6,01180d63797c8e?readDeleted=true: 404 Not Found

**System Setup**

OS: Ubuntu 20.04 
weed version 30GB 2.33 91a3ac9 linux amd64

Server 1:
nohup ./weed -logdir=/home/bf/weed/log/master  master -ip.bind=10.1.2.3 -ip=10.1.2.3 -port=6200 &
nohup ./weed -logdir=/home/bf/weed/log/vol1 volume -dir=/mnt/sda/data/weed -max=0 -mserver=10.1.2.3:6200  -port=9090 -ip=[2a01:4f8:236:60d3::8]  -ip.bind=[2a01:4f8:236:60d3::8] &
nohup ./weed -logdir=/home/bf/weed/log/vol2 volume -dir=/mnt/sdb/data/weed -max=0 -mserver=10.1.2.3:6200  -port=9090 -ip=[2a01:4f8:236:60d3::9]  -ip.bind=[2a01:4f8:236:60d3::9]&
nohup ./weed -logdir=/home/bf/weed/log/filer  filer -encryptVolumeData -collection="light_data" -master=10.1.2.3:6200 -ip.bind=10.1.2.3 -ip=10.1.2.3 -port=6400 -peers=10.1.2.5:6400&

server2:

nohup ./weed -logdir=/home/bf/weed/log/vol1 volume -dir=/bfdata/weed -max=0 -mserver=10.1.2.3:6200 -port=6300 -ip=46.24.152.75 -ip.bind=46.24.152.75 &
nohup ./weed -logdir=/home/bf/weed/log/vol2 volume -dir=/data/bfdata/weed -max=0 -mserver=10.1.2.3:6200 -port=6301 -ip=46.24.152.75 -ip.bind=46.24.152.75 &
nohup ./weed -logdir=/home/bf/weed/log/filer  filer -encryptVolumeData -collection="light_data" -master=10.1.2.3:6200 -ip.bind=10.1.2.5 -ip=10.1.2.5 -port=6400 -peers=10.1.2.3:6400&

**Expected behavior**
The data should not be lost.

**Comment by kmlebedev** [comment]
@chaos369 plz show `echo volume.list | weed shell`

**Comment by chrislusf** [comment]
@chaos369 any update?

**Comment by chaos369** [comment]
> volume.list
Topology volumeSizeLimit:30000 MB hdd(volume:84/303 active:84 free:219 remote:0)
  DataCenter DefaultDataCenter hdd(volume:84/303 active:84 free:219 remote:0)
    Rack DefaultRack hdd(volume:84/303 active:84 free:219 remote:0)
      DataNode 46.4.52.175:6300 hdd(volume:9/59 active:9 free:50 remote:0)
        Disk hdd(volume:9/59 active:9 free:50 remote:0)
          volume id:2 size:168142440 collection:"file_info" file_count:2200 delete_count:2192 deleted_byte_count:228490 version:3 compact_revision:6 modified_at_second:1615996457
          volume id:3 size:537231760 collection:"file_info" file_count:2124 delete_count:2107 deleted_byte_count:219194 version:3 compact_revision:4 modified_at_second:1615996457
          volume id:4 size:8 collection:"file_info" version:3 compact_revision:5 modified_at_second:1615971026
          volume id:8 size:3840 collection:"light_data" file_count:2 version:3 modified_at_second:1615960826
          volume id:9 size:4536240 collection:"light_data" file_count:10 version:3 modified_at_second:1615976281
          volume id:11 size:7504 collection:"light_data" file_count:4 version:3 modified_at_second:1615976401
          volume id:13 size:29896 collection:"light_data" file_count:7 version:3 modified_at_second:1615978621
          volume id:14 size:8312 collection:"light_data" file_count:4 version:3 modified_at_second:1615968806
          volume id:68 size:720 collection:"light_data" file_count:1 version:3 modified_at_second:1615989176
        Disk hdd total size:709960720 file_count:4352 deleted_file:4299 deleted_bytes:447684
      DataNode 46.4.52.175:6300 total size:709960720 file_count:4352 deleted_file:4299 deleted_bytes:447684
      DataNode 46.4.52.175:6301 hdd(volume:7/61 active:7 free:54 remote:0)
        Disk hdd(volume:7/61 active:7 free:54 remote:0)
          volume id:1 size:40 collection:"file_info" delete_count:1 version:3 compact_revision:6 modified_at_second:1615971027
          volume id:5 size:368 collection:"file_info" file_count:1 version:3 compact_revision:5 modified_at_second:1615971027
          volume id:6 size:684205936 collection:"file_info" file_count:2161 delete_count:2139 deleted_byte_count:222414 version:3 compact_revision:6 modified_at_second:1615996457
          volume id:7 size:503823480 collection:"file_info" file_count:2276 delete_count:2260 deleted_byte_count:360264 version:3 compact_revision:5 modified_at_second:1615996457
          volume id:10 size:13904 collection:"light_data" file_count:6 version:3 modified_at_second:1615968866
          volume id:12 size:8076368 collection:"light_data" file_count:10 version:3 modified_at_second:1615978321
          volume id:65 size:1032 collection:"light_data" file_count:1 version:3 modified_at_second:1615983331
        Disk hdd total size:1196121128 file_count:4455 deleted_file:4400 deleted_bytes:582678
      DataNode 46.4.52.175:6301 total size:1196121128 file_count:4455 deleted_file:4400 deleted_bytes:582678
      DataNode [2a01:4f8:191:60d3::8]:9090 hdd(volume:26/91 active:26 free:65 remote:0)
        Disk hdd(volume:26/91 active:26 free:65 remote:0)
          volume id:1 size:31471469120 collection:"file_info" file_count:971 delete_count:153 deleted_byte_count:3509045540 version:3 modified_at_second:1614817501
          volume id:4 size:31510948832 collection:"file_info" file_count:951 delete_count:122 deleted_byte_count:3560130509 version:3 modified_at_second:1615292475
          volume id:5 size:31864798560 collection:"file_info" file_count:921 delete_count:134 deleted_byte_count:3532890854 version:3 modified_at_second:1614817502
          volume id:8 size:609800 collection:"light_data" file_count:219 version:3 modified_at_second:1615773977
          volume id:10 size:391912 collection:"light_data" file_count:228 version:3 modified_at_second:1615828277
          volume id:14 size:643736 collection:"light_data" file_count:232 version:3 modified_at_second:1615819397
          volume id:16 size:31520144072 collection:"file_info" file_count:960 delete_count:82 deleted_byte_count:2936217122 version:3 compact_revision:1 modified_at_second:1614532526
          volume id:17 size:31692727968 collection:"file_info" file_count:667 delete_count:171 deleted_byte_count:7991888103 version:3 modified_at_second:1610602822
          volume id:21 size:31502142848 collection:"file_info" file_count:971 delete_count:83 deleted_byte_count:3461088557 version:3 compact_revision:1 modified_at_second:1614532521
          volume id:22 size:31515314408 collection:"file_info" file_count:1387 delete_count:143 deleted_byte_count:1862427294 version:3 compact_revision:1 modified_at_second:1615080851
          volume id:25 size:31493635328 collection:"file_info" file_count:1024 delete_count:164 deleted_byte_count:9228202949 version:3 compact_revision:1 modified_at_second:1614532531
          volume id:28 size:31638389776 collection:"file_info" file_count:1082 delete_count:238 deleted_byte_count:9463664070 version:3 modified_at_second:1614532511
          volume id:29 size:31799614784 collection:"file_info" file_count:724 delete_count:144 deleted_byte_count:7484332231 version:3 compact_revision:2 modified_at_second:1615442195
          volume id:30 size:31581452672 collection:"file_info" file_count:737 delete_count:166 deleted_byte_count:7738503871 version:3 compact_revision:2 modified_at_second:1615442198
          volume id:31 size:31651877592 collection:"file_info" file_count:757 delete_count:160 deleted_byte_count:7795955101 version:3 compact_revision:2 modified_at_second:1615442198
          volume id:32 size:31608218824 collection:"file_info" file_count:724 delete_count:143 deleted_byte_count:6773047195 version:3 compact_revision:2 modified_at_second:1615442198
          volume id:35 size:31537319224 collection:"file_info" file_count:765 delete_count:96 deleted_byte_count:4338641986 version:3 compact_revision:3 modified_at_second:1615976355
          volume id:51 size:31565681240 collection:"file_info" file_count:1126 delete_count:252 deleted_byte_count:5523106987 version:3 compact_revision:2 modified_at_second:1615543743
          volume id:53 size:31554216376 collection:"file_info" file_count:1162 delete_count:263 deleted_byte_count:5250429809 version:3 compact_revision:2 modified_at_second:1614881746
          volume id:54 size:31532985048 collection:"file_info" file_count:1096 delete_count:241 deleted_byte_count:5981436198 version:3 compact_revision:2 modified_at_second:1615543749
          volume id:55 size:31481598048 collection:"file_info" file_count:1220 delete_count:291 deleted_byte_count:6170727277 version:3 compact_revision:2 modified_at_second:1614881746
          volume id:56 size:31488301728 collection:"file_info" file_count:1245 delete_count:262 deleted_byte_count:5555613639 version:3 compact_revision:2 modified_at_second:1614881734
          volume id:58 size:12867625088 collection:"file_info" file_count:5298 delete_count:2345 deleted_byte_count:1670905628 version:3 compact_revision:3 modified_at_second:1615996457
          volume id:62 size:14397148664 collection:"file_info" file_count:5259 delete_count:2317 deleted_byte_count:1420191794 version:3 compact_revision:2 modified_at_second:1615996457
          volume id:63 size:12364170288 collection:"file_info" file_count:5237 delete_count:2318 deleted_byte_count:1282060708 version:3 compact_revision:4 modified_at_second:1615996457
          volume id:66 size:4120 collection:"light_data" file_count:1 version:3 modified_at_second:1615987796
        Disk hdd total size:639641430056 file_count:34964 deleted_file:10288 deleted_bytes:112530507422
      DataNode [2a01:4f8:191:60d3::8]:9090 total size:639641430056 file_count:34964 deleted_file:10288 deleted_bytes:112530507422
      DataNode [2a01:4f8:191:60d3::9]:9090 hdd(volume:42/92 active:42 free:50 remote:0)
        Disk hdd(volume:42/92 active:42 free:50 remote:0)
          volume id:2 size:31609142800 collection:"file_info" file_count:931 delete_count:122 deleted_byte_count:3158762432 version:3 compact_revision:1 modified_at_second:1614590328
          volume id:3 size:31543971280 collection:"file_info" file_count:934 delete_count:144 deleted_byte_count:3594606065 version:3 modified_at_second:1614817502
          volume id:6 size:31583580496 collection:"file_info" file_count:936 delete_count:122 deleted_byte_count:2955320288 version:3 compact_revision:1 modified_at_second:1613093948
          volume id:7 size:31520790888 collection:"file_info" file_count:775 delete_count:124 deleted_byte_count:3322490100 version:3 modified_at_second:1614589759
          volume id:9 size:898776 collection:"light_data" file_count:267 version:3 modified_at_second:1615830378
          volume id:11 size:877728 collection:"light_data" file_count:285 version:3 modified_at_second:1615826057
          volume id:12 size:894152 collection:"light_data" file_count:266 version:3 modified_at_second:1615904898
          volume id:13 size:516368 collection:"light_data" file_count:261 version:3 modified_at_second:1615828577
          volume id:15 size:31472192152 collection:"file_info" file_count:679 delete_count:174 deleted_byte_count:8024514417 version:3 modified_at_second:1610705280
          volume id:18 size:31742394744 collection:"file_info" file_count:699 delete_count:189 deleted_byte_count:9403948001 version:3 modified_at_second:1612708131
          volume id:19 size:31981391080 collection:"file_info" file_count:700 delete_count:177 deleted_byte_count:8534118790 version:3 modified_at_second:1611666815
          volume id:20 size:31504485088 collection:"file_info" file_count:672 delete_count:200 deleted_byte_count:8902124081 version:3 modified_at_second:1612346358
          volume id:23 size:31598202848 collection:"file_info" file_count:1074 delete_count:232 deleted_byte_count:8805866284 version:3 modified_at_second:1614863152
          volume id:24 size:31507304208 collection:"file_info" file_count:1064 delete_count:162 deleted_byte_count:8729760727 version:3 compact_revision:1 modified_at_second:1614532529
          volume id:26 size:31521783976 collection:"file_info" file_count:1102 delete_count:93 deleted_byte_count:2884771625 version:3 compact_revision:1 modified_at_second:1614863152
          volume id:27 size:31482496848 collection:"file_info" file_count:1408 delete_count:160 deleted_byte_count:2219291890 version:3 compact_revision:1 modified_at_second:1615075398
          volume id:33 size:32173104848 collection:"file_info" file_count:761 delete_count:185 deleted_byte_count:7981310981 version:3 compact_revision:2 modified_at_second:1615442196
          volume id:34 size:31484348224 collection:"file_info" file_count:769 delete_count:173 deleted_byte_count:7506365518 version:3 compact_revision:2 modified_at_second:1615442194
          volume id:36 size:31518391616 collection:"file_info" file_count:1446 delete_count:94 deleted_byte_count:2076955354 version:3 compact_revision:1 modified_at_second:1615543749
          volume id:37 size:31578887416 collection:"file_info" file_count:1600 delete_count:411 deleted_byte_count:7162574667 version:3 modified_at_second:1615543748
          volume id:38 size:31520418032 collection:"file_info" file_count:1606 delete_count:431 deleted_byte_count:8698824157 version:3 modified_at_second:1613901400
          volume id:39 size:31572288696 collection:"file_info" file_count:1608 delete_count:343 deleted_byte_count:7047754706 version:3 modified_at_second:1615575793
          volume id:40 size:31492769664 collection:"file_info" file_count:1382 delete_count:56 deleted_byte_count:1460428433 version:3 compact_revision:1 modified_at_second:1615543749
          volume id:41 size:31684470312 collection:"file_info" file_count:1618 delete_count:416 deleted_byte_count:7652093493 version:3 modified_at_second:1614783177
          volume id:42 size:31942378120 collection:"file_info" file_count:1673 delete_count:487 deleted_byte_count:9179508105 version:3 modified_at_second:1615575793
          volume id:43 size:31489203840 collection:"file_info" file_count:844 delete_count:367 deleted_byte_count:9368186160 version:3 compact_revision:1 modified_at_second:1615543743
          volume id:44 size:31474895320 collection:"file_info" file_count:879 delete_count:231 deleted_byte_count:3680991320 version:3 compact_revision:2 modified_at_second:1615992860
          volume id:45 size:31505283224 collection:"file_info" file_count:845 delete_count:232 deleted_byte_count:4157352588 version:3 compact_revision:2 modified_at_second:1615075466
          volume id:46 size:31554829360 collection:"file_info" file_count:904 delete_count:253 deleted_byte_count:4346066652 version:3 compact_revision:2 modified_at_second:1615075489
          volume id:47 size:31563074832 collection:"file_info" file_count:848 delete_count:321 deleted_byte_count:8148030830 version:3 compact_revision:1 modified_at_second:1615366685
          volume id:48 size:31537833864 collection:"file_info" file_count:922 delete_count:244 deleted_byte_count:4778410977 version:3 compact_revision:2 modified_at_second:1615075484
          volume id:49 size:31484742072 collection:"file_info" file_count:938 delete_count:354 deleted_byte_count:8020017930 version:3 compact_revision:1 modified_at_second:1615217135
          volume id:50 size:31467863848 collection:"file_info" file_count:1189 delete_count:253 deleted_byte_count:5030070266 version:3 compact_revision:2 modified_at_second:1615543743
          volume id:52 size:31473012616 collection:"file_info" file_count:1180 delete_count:232 deleted_byte_count:5435115713 version:3 compact_revision:2 modified_at_second:1614881746
          volume id:57 size:11986382912 collection:"file_info" file_count:3214 delete_count:118 deleted_byte_count:1450498368 version:3 compact_revision:3 modified_at_second:1615996456
          volume id:59 size:12535314704 collection:"file_info" file_count:3148 delete_count:114 deleted_byte_count:1356169235 version:3 compact_revision:2 modified_at_second:1615996456
          volume id:60 size:11825120136 collection:"file_info" file_count:3026 delete_count:119 deleted_byte_count:1333125542 version:3 compact_revision:3 modified_at_second:1616011043
          volume id:61 size:11887474080 collection:"file_info" file_count:2999 delete_count:14 deleted_byte_count:67293114 version:3 compact_revision:3 modified_at_second:1616011537
          volume id:64 size:8 collection:"light_data" version:3 modified_at_second:1615980991
          volume id:67 size:4205680 collection:"light_data" file_count:6 version:3 modified_at_second:1615996435
          volume id:69 size:6519776 collection:"light_data" file_count:5 version:3 modified_at_second:1616011077
          volume id:70 size:421224 collection:"light_data" file_count:2 version:3 modified_at_second:1616011557
        Disk hdd total size:995834157856 file_count:45465 deleted_file:7347 deleted_bytes:186472718809
      DataNode [2a01:4f8:191:60d3::9]:9090 total size:995834157856 file_count:45465 deleted_file:7347 deleted_bytes:186472718809
    Rack DefaultRack total size:1637381669760 file_count:89236 deleted_file:26334 deleted_bytes:299004256593
  DataCenter DefaultDataCenter total size:1637381669760 file_count:89236 deleted_file:26334 deleted_bytes:299004256593
total size:1637381669760 file_count:89236 deleted_file:26334 deleted_bytes:299004256593

**Comment by chaos369** [comment]
When I access a file which was lost. 

==> filer/weed.ERROR <==
E0318 01:38:22 37939 stream.go:42] read chunk: fetch http://[2a01:4f8:236:60d3::9]:9090/6,0152541bd228b8?readDeleted=true: http://[2a01:4f8:236:60d3::9]:9090/6,0152541bd228b8?readDeleted=true: 404 Not Found


> fs.cat f/01/01eeecc388b81c4c555430cf7c4069e3dcb674b9ca01ef7723ca16a669398fd4
{"Server":"46.24.152.75:6300","FileID":"3,0152358eb8f2a1","FileSize":2006433468,"Reference":1}>


root@Ubuntu-2004:~/tmp# wget http://46.24.152.75:6300/3,0152358eb8f2a1
--2021-03-18 01:40:16--  http://46.24.152.75:6300/3,0152358eb8f2a1
Connecting to 46.24.152.75:6300... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2006433468 (1.9G) [video/mp4]
Saving to: ‘3,0152358eb8f2a1’

3,0152358eb8f2a1               3%[>                                              ]  64.00M   112MB/s    in 0.6s

2021-03-18 01:40:16 (112 MB/s) - Connection closed at byte 67108947. Retrying.

--2021-03-18 01:40:17--  (try: 2)  http://46.24.152.75:6300/3,0152358eb8f2a1
Connecting to 46.24.152.75:6300... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 2006433468 (1.9G), 1939324521 (1.8G) remaining [video/mp4]
Saving to: ‘3,0152358eb8f2a1’

3,0152358eb8f2a1               3%[+                                              ]  64.00M  --.-KB/s    in 0s

2021-03-18 01:40:17 (7.94 MB/s) - Connection closed at byte 67109030. Retrying.

--2021-03-18 01:40:19--  (try: 3)  http://46.24.152.75:6300/3,0152358eb8f2a1
Connecting to 46.24.152.75:6300... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 2006433468 (1.9G), 1939324438 (1.8G) remaining [video/mp4]
Saving to: ‘3,0152358eb8f2a1’

3,0152358eb8f2a1               3%[+                                              ]  64.00M  --.-KB/s    in 0s

2021-03-18 01:40:19 (8.40 MB/s) - Connection closed at byte 67109113. Retrying.

**Comment by chaos369** [comment]
I checked the lost files on the server1, they were successfully retrieved.

**Comment by chrislusf** [comment]
There are two volume 9 in the volume.list
```
       DataNode 46.4.52.175:6300 hdd(volume:9/59 active:9 free:50 remote:0)
         volume id:9 size:4536240 collection:"light_data" file_count:10 version:3 modified_at_second:1615976281
       DataNode [2a01:4f8:191:60d3::9]:9090 hdd(volume:42/92 active:42 free:50 remote:0)
         volume id:9 size:898776 collection:"light_data" file_count:267 version:3 modified_at_second:1615830378
```

Seems there are already some data on server2 before merging them into the same cluster.

**Comment by chaos369** [comment]
Sorry,  2a01:4f8:191:60d3::9 is 2a01:4f8:236:60d3::9, I just replaced to hide the real ip address. only one volume ::9. 

And yes, There're some testing data. Here're the steps of merging.
0 There's an old S0 with 1 master(127.0.0.1:9333), 1 filer (127.0.0.1:8888), 3 volumes(volume_ip:port) on server A.

1 I set up the new S1 with 1 master(10.1.2.3:6000), 1 filer(10.1.2.3:6400) on server A, 3 volumes(volume_ip1:port1), S2 with 1 filer, 2 volumes on server B.

2 I tested files uploading and downloading well on S1 and S2.

3 I  made a soft link to the S0' volume data path for S1' volume.

4 I stoped S0, and started S1.

5 I found some data could not retrived, so I modify the S1's volume data path to S0's data path directly.

6 Still there're some files can't be retrived.
